File: app/src/main/java/com/example/mrakopediareader/api/dto/Category.java
package com.example.mrakopediareader.api.dto;

public class Category {
    private final String title;

    private final String url;

    public Category(String title, String url) {
        this.title = title;
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public String getUrl() {
        return url;
    }
}

File: app/src/main/java/com/example/mrakopediareader/api/dto/Category.kt
package com.example.mrakopediareader.api.dto

data class Category(val title: String, val url: String) {
    fun serialize(): String {
        return "$title${JOINER}$url"
    }

    companion object {
        private const val JOINER = "|"

        fun parse(source: String): Category {
            val (title, url) = source.split(JOINER)
            return Category(title, url)
        }
    }
}